providers/standard/src/airflow/providers/standard/decorators/python_virtualenv.py

```diff
@@ -18,12 +18,22 @@
 
 from typing import TYPE_CHECKING, Callable
 
-from airflow.decorators.base import task_decorator_factory
+from airflow.providers.standard.version_compat import AIRFLOW_V_3_0_PLUS
+
+if AIRFLOW_V_3_0_PLUS:
+    from airflow.sdk.bases.decorator import task_decorator_factory
+else:
+    from airflow.decorators.base import task_decorator_factory
+
+
 from airflow.providers.standard.decorators.python import _PythonDecoratedOperator
 from airflow.providers.standard.operators.python import PythonVirtualenvOperator
 
 if TYPE_CHECKING:
-    from airflow.decorators.base import TaskDecorator
+    if AIRFLOW_V_3_0_PLUS:
+        from airflow.sdk.bases.decorator import TaskDecorator
+    else:
+        from airflow.decorators.base import TaskDecorator
```

Comment:
I don't think conditions really works in type checking blocks, so elsewhere we have just gone with 

```suggestion
    from airflow.sdk.bases.decorator import TaskDecorator

```